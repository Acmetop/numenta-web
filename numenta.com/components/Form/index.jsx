// Numenta company website source code. Copyright Â© 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import React from 'react'

import styles from './index.css'


const Form = ({action, children, name, onSubmit}) => (
  <form
    acceptCharset="UTF-8"
    action={action}
    autoComplete="off"
    className={styles.form}
    encType="multipart/form-data"
    id={name}
    method="post"
    name={name}
    onSubmit={onSubmit}
  >
    {children}
  </form>
)

Form.propTypes = {
  action: React.PropTypes.string,
  children: React.PropTypes.node.isRequired,
  name: React.PropTypes.string,
  onSubmit: React.PropTypes.func,
}

export default Form
