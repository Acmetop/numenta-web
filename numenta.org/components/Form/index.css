/**
 * Numenta.org HTM Community website source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

.form {
  composes: pa0 from 'tachyons';
  composes: mb3 from 'tachyons';
  composes: mh0 from 'tachyons';
  composes: mt0 from 'tachyons';
}
