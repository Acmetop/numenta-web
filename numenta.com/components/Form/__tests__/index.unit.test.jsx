import React from 'react'
import renderer from 'react-test-renderer'

import Form from '../../Form'


describe('<Form /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Form />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
