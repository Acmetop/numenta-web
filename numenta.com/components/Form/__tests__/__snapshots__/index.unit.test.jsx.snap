exports[`<Form /> React component unit test suite Renders correctly 1`] = `
<form
  acceptCharset="UTF-8"
  action={undefined}
  autoComplete="off"
  className="form"
  encType="multipart/form-data"
  id={undefined}
  method="post"
  name={undefined}
  onSubmit={undefined} />
`;
